### PR TITLE
Mapping

### PR DESCRIPTION
**Pull request by Iaxama**
Main changes:
 - added BOX events with its drawer to handle bounding boxes
 - Added BoxConverter module to transform BOX events using extrinsic camera parameters. (probably the new drawer with 6DOF calibration would do the same thing better)
 - Optimization of DualCamTransform module (same comment as above)
 - Some changes to autosaccade module to use velocity control for saccade motion in order to make sharper movements

### CODE REVIEW

File: documentation/vMapping_tutorial.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+#vMapping App
```
Comment (arrenglover) [inline documentation/vMapping_tutorial.md]: this app is probably redundant and we want to make sure undistortion can happen with the new module - tested on the robot!

File: lib/include/event-driven/vCodec.h [hunk 2]
```diff
@@ -337,9 +337,32 @@ template <> inline int countTime< std::vector<int32_t> > (const std::vector<int3
 }
 
 
+/// \brief an AddressEvent with a width and height to draw bounding boxes
```
Comment (arrenglover) [inline lib/include/event-driven/vCodec.h]: we'll need to update the class to be consistent with the other events

File: src/applications/CMakeLists.txt [hunk 3]
```diff
@@ -6,6 +6,7 @@ option(ENABLE_calibrate "Build camera calibration" OFF)
 option(ENABLE_view "Build basic viewer" ON)
 option(ENABLE_corner "Build corner detector" OFF)
 option(ENABLE_dualCamTransform "Build event to frame transform" OFF)
+option(ENABLE_vAnnotation "Build application to automatically annotate events" OFF)
```
Comment (arrenglover) [inline src/applications/CMakeLists.txt]: remove vAnnotation from event-driven and put it as an external module with all the code for learning object recognition

File: src/processing/BoxConverter/BoxConverter.xml [hunk 4]
```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
```
Comment (arrenglover) [inline src/processing/BoxConverter/BoxConverter.xml]: move to external repo

File: src/processing/vFramer/src/other_drawers.cpp [hunk 5]
```diff
@@ -310,3 +310,43 @@ void saeDraw::draw(cv::Mat &image, const ev::vQueue &eSet, int vTime)
...
+}
+
+std::string boxDraw::getEventType()
+{
+    return BoxEvent::tag;
+}
+
+void boxDraw::draw(cv::Mat &image, const ev::vQueue &eSet, int vTime)
+{
+    cv::Scalar blue = CV_RGB(0, 0, 255);
```
Comment (arrenglover) [inline src/processing/vFramer/src/other_drawers.cpp]: use a predefined colour in the vDraw base class

File: src/processing/vCorner/vCorner.xml [hunk 6]
```diff
@@ -15,14 +15,18 @@
     <arguments>
         <param desc="Specifies the stem name of ports created by the module." default="vCorner"> name </param>
         <param desc="Sets ouput port to use strict protocol." default="true"> strict </param>
-        <param desc="Number of pixels on the x-axis of the sensor." default="128"> width </param>
-        <param desc="Number of pixels on the y-axis of the sensor." default="128"> height </param>
-        <param desc="Size of Sobel filter used for the detection." default="5"> filterSize </param>
-        <param desc="Length of the temporal window in seconds used to accumulate events." default="0.01"> tempsize </param>
+        <param desc="Number of pixels on the x-axis of the sensor." default="304"> width </param>
```
Comment (arrenglover) [inline src/processing/vCorner/vCorner.xml]: double check that this file isn't actually in the pull request. If it's in the pull request, remove it (make it a new pull request). If it's not just leave this

**Comment by arrenglover** [comment]
bump @Iaxama

**Comment by arrenglover** [comment]
@Iaxama - are any of these changes worthwhile anymore? Is it easier to just delete this branch and redo what we need later? is there a demo based on this branch?

**Comment by Iaxama** [comment]
> @Iaxama - are any of these changes worthwhile anymore? Is it easier to just delete this branch and redo what we need later? is there a demo based on this branch?

No I think is safe to just delete this branch. There is nothing major and the features that were implemented have been replaced or have become obsolete. It would take less effort to reimplement any feature that we think useful in the future rather than trying to get anything useful out of this

**Pull request closed without merging**